google/cloud/aiplatform/model_monitoring/objective.py

```diff
@@ -0,0 +1,287 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import abc
+from typing import Optional, Dict
+
+from google.cloud.aiplatform_v1.types import (
+    io as gca_io,
+    ThresholdConfig as gca_threshold_config,
+    model_monitoring as gca_model_monitoring,
+)
+
+
+class _SkewDetectionConfig(abc.ABC):
```

Comment:
Can we revisit the decision to add these configs as replacement to the proto configs? How is the usability improved vs using the proto configs? I wonder if this will add too much overhead vs just aliasing the proto configs. 

cc: @qijing93

Reply:
One argument I have in favor of adding these higher-level abstracted classes is that it allows for error checking on the data set. According to the documentation in the proto files, `data_format` is only an applicable field if the data source is from GCS. So the currently error behavior that I implemented in the SDK looks like this:
- if `data_source = "gs://..."` and `data_format not in VALID_SET`, then the code throws an actionable error to the user
- if `data_source = "bq://..." (or something else)`, then `data_format` is ignored even if it's specified

Other than this additional error check, I don't really have any arguments for or against having this extra abstraction layer. Wrapping GAPIC config classes into these new classes was OK'd during the design doc review, so I just stuck with it during the implementation.

Reply:
Just thought of another argument for creating the SDK class for these configs: this makes more sense idiomatically because we're also creating an MDM job class, which is a class in the SDK. I think without these config classes, developers will end up mixing-and-matching SDK classes and GAPIC classes when they want to create MDM jobs and configs, which is not really ideal from a DevEx point of view.

Reply:
I assume the service would validate the error case above. @qijing93 to confirm.

We can alias the GAPIC classes into the SDK namespace.

I'm open either way. The main downside is that we will need to maintain these configs and they can become out of sync with the GAPIC configs. This is an issue we've run into in other places in the SDK.

It would be great to see an example of the improved UX of these configs vs the GAPIC configs to see if it's worth the trade off.

Reply:
Yes the service should validate the error case above.